snapcraft/_store.py

```diff
@@ -793,19 +793,71 @@ def validate(snap_name, validations, revoke=False, key=None):
         if revoke:
             assertion['revoked'] = "true"
 
-        assertion = _sign_validation(validation, assertion, key)
+        assertion = _sign_assertion(validation, assertion, key, 'validations')
 
         # Save assertion to a properly named file
         fname = '{}-{}-r{}.assertion'.format(snap_name, gated_name, rev)
         with open(fname, 'wb') as f:
             f.write(assertion)
 
-        store.push_validation(snap_id, assertion)
+        store.push_assertion(snap_id, assertion, endpoint='validations')
 
 
 validation_re = re.compile('^[^=]+=[0-9]+$')
 
 
+def collaborate(snap_name, key):
+    store = storeapi.StoreClient()
+
+    with _requires_login():
+        account_info = store.get_account_information()
+
+    release = storeapi.constants.DEFAULT_SERIES
+    try:
+        snap_id = account_info['snaps'][release][snap_name]['snap-id']
+    except KeyError:
+        raise storeapi.errors.SnapNotFoundError(snap_name)
+    developers = _get_developers(snap_id)
+
+    # XXX: Do the amendments via UI here.
+    #
+    # The data will look like:
+    # {'snap_developer': {
+    #      'type: 'snap-developer',
+    #      'authority-id': <account_id of the publisher Or `canonical`>,
+    #      'publisher-id': <account_id of the publisher>,
+    #      'snap-id': 'snap_id',
+    #      'developers': [{
+    #          'developer-id': 'account_id of dev-1',
+    #          'since': '2017-02-10T08:35:00.390258Z'
+    #         },{
+    #          'developer-id': 'account_id of dev-2',
+    #          'since': '2017-02-10T08:35:00.390258Z',
+    #          'until': '2018-02-10T08:35:00.390258Z'
+    #         }],
+    #      }
+    # }
+
+    _sign_developers(snap_id, developers['snap_developer'], key)
+
+
+def _get_developers(snap_id):
+    store = storeapi.StoreClient()
+    developers = {'snap_developer': []}
+    try:
+        developers = store.get_assertion(snap_id, 'developers')
+    except storeapi.errors.StoreValidationError as e:
+        if e.error_list[0]['code'] != 'snap-developer-not-found':
+            raise
```

Comment:
I think it would be a little clearer like this:

    try:
        return store.get_assertion(snap_id, 'developers')
    except storeapi.errors.StoreValidationError as e:
        if e.error_list[0]['code'] == 'snap-developer-not-found':
            return {'snap_developer': []}
        else:
            raise